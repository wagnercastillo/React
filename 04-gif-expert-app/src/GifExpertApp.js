import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({ defaultCategorias = [] }) => {
	const [categorias, setCategorias] = useState(defaultCategorias);

	return (
		<>
			<h2>GifExpertApp</h2>
			<AddCategory setCategorias={setCategorias} />
			<hr />

			<ol>
				{categorias.map((categoria) => (
					<GifGrid key={categoria} categoria={categoria} />
				))}
			</ol>
		</>
	);
};

export default GifExpertApp;
