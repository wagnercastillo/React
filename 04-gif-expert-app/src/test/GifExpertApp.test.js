import shallow from 'enzyme/shallow';
import React from 'react';

import GifExpertApp from "../GifExpertApp";

describe("Pruebas en el componente <GifExpertApp />", () => {
	
    test("Debe mostrar correctamente el componente", () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorias', () => { 

        const categories = ['One Punch', 'Samurai X'];
        const wrapper = shallow(<GifExpertApp defaultCategorias={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);



     })

});
