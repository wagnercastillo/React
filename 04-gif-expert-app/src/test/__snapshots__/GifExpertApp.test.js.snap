// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="One Punch"
      key="One Punch"
    />
    <GifGrid
      categoria="Samurai X"
      key="Samurai X"
    />
  </ol>
</Fragment>
`;
